src/config.ts

```diff
@@ -55,3 +44,51 @@ export async function getConfig(flags: Flags) {
   config = Object.assign({}, config, strippedFlags);
   return config;
 }
+
+const validConfigExtensions = ['.js', '.mjs', '.cjs', '.json'];
+type ConfigExtensions = typeof validConfigExtensions[number];
+
+async function parseConfigFile(configPath: string): Promise<Flags> {
+  const typeOfConfig = getTypeOfConfig(configPath);
+
+  switch (typeOfConfig) {
+    case '.json':
+      return readJsonConfigFile(configPath);
+    case '.js':
+    case '.mjs':
+    case '.cjs':
+      return importConfigFile(configPath);
+  }
+
+  throw new Error(`Config file ${configPath} is invalid`);
+}
+
+function getTypeOfConfig(configPath: string): ConfigExtensions {
+  // Returning json in case file doesn't have an extension for backward compatibility
+  const configExtension = path.extname(configPath) || '.json';
+
+  if (validConfigExtensions.includes(configExtension)) {
+    return configExtension as ConfigExtensions;
+  }
+
+  throw new Error(
+    `Config file should be either of extensions ${validConfigExtensions.join(
+      ','
+    )}`
+  );
+}
+
+async function importConfigFile(configPath: string): Promise<Flags> {
+  const config = (await import(path.resolve(process.cwd(), configPath)))
```

Comment:
Should we be doing a `path.join` here instead to get an absolute path to the config file?

Reply:
Well I don't have a choice here, I thought since this is import statement, so path.resolve would be a better one but tried once with path.join & it's working as same. I've changed it to path.join for now all tests seems to be passing in my local. I've made other changes also suggested by you to the PR please check

Reply:
reverting this to `path.resolve()`, I remember now why I was using path.resolve, it was to support absolute paths of config file.

path.resolve combines paths till an absolute path isn't defined. It does the resolution from right to left. So configPath being at the right, if the user passes an absolute path ignoring left argument (cwd) but if the user sends passes a relative one it would be resolved absolutely by combining working directory (cwd)